hls4ml/templates/quartus/firmware/nnet_utils/nnet_activation.h

```diff
@@ -99,12 +99,13 @@ void  relu1(data_T data[CONFIG_T::n_in], res_T res[CONFIG_T::n_in])
 template<class data_T, class res_T, typename CONFIG_T>
 void  sigmoid(data_T data[CONFIG_T::n_in], res_T res[CONFIG_T::n_in])
 {
+    int MAX_TABLE_value=8;
     #include "activation_tables/sigmoid_table.tb"
     // Index into the lookup table based on data
     #pragma unroll
     for (int ii=0; ii<CONFIG_T::n_in; ii++) {
         int data_round = (data[ii]*(CONFIG_T::table_size/16)).to_int();
-        int index = data_round + 8*CONFIG_T::table_size/16;
+        int index = data_round + MAX_TABLE_value *CONFIG_T::table_size/16;
```

Comment:
Should also include the minimum value and `table_size/16` should become `table_size/(MAX_TABLE_value-MIN_TABLE_value)` same applies in other places too where `table_size/16` is used.

Reply:
I will soon rewrite all code for activation function with the optimisation I made so for now,I will let like this